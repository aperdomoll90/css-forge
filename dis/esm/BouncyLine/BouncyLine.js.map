{"version":3,"file":"BouncyLine.js","sourceRoot":"","sources":["../../../src/BouncyLine/BouncyLine.tsx"],"names":[],"mappings":";;;;;;;;;;;;AAAA,OAAO,EAAE,MAAM,EAAE,QAAQ,EAAE,MAAM,OAAO,CAAA;AACxC,OAAO,EAAE,IAAI,EAAE,OAAO,EAAE,MAAM,MAAM,CAAA;AACpC,OAAO,cAAc,CAAA;AACrB,OAAO,2BAA2B,CAAA;AAElC,MAAM,CAAC,IAAM,UAAU,GAAiB;IAChC,IAAA,KAAsB,QAAQ,CAAC,EAAE,CAAC,EAAE,CAAC,EAAE,CAAC,EAAE,CAAC,EAAE,CAAC,EAA7C,MAAM,QAAA,EAAE,SAAS,QAA4B,CAAA;IACpD,IAAM,GAAG,GAAG,MAAM,CAAiB,IAAI,CAAC,CAAA;IAExC,IAAM,cAAc,GAAG,UAAC,CAAmC;QACzD,IAAI,GAAG,IAAI,GAAG,CAAC,OAAO,EAAE;YACtB,IAAM,KAAK,GAAG,GAAG,CAAC,OAAO,CAAC,WAAW,CAAA;YACrC,IAAM,MAAM,GAAG,GAAG,CAAC,OAAO,CAAC,YAAY,CAAA;YACvC,IAAM,IAAI,GAAG,CAAC,CAAC,CAAC,WAAW,CAAC,OAAO,GAAG,KAAK,CAAC,GAAG,GAAG,CAAA;YAClD,IAAM,IAAI,GAAG,CAAC,CAAC,CAAC,WAAW,CAAC,OAAO,GAAG,MAAM,CAAC,GAAG,GAAG,CAAA;YACnD,SAAS,CAAC;gBACR,CAAC,EAAE,IAAI;gBACP,CAAC,EAAE,IAAI;aACR,CAAC,CAAA;YACF,UAAU,CAAC,KAAK,CAAC,CAAA;SAClB;IACH,CAAC,CAAA;IACD,IAAM,UAAU,GAAG,UAAC,KAAa;QAC/B,IAAM,IAAI,GAAG,QAAQ,CAAC,aAAa,CAAC,MAAM,CAAmB,CAAA;QACvD,IAAA,CAAC,GAAQ,MAAM,EAAd,EAAE,CAAC,GAAK,MAAM,EAAX,CAAW;QACrB,IAAI,CAAC,YAAY,CAAC,GAAG,EAAE,kBAAW,CAAC,cAAI,CAAC,aAAU,CAAC,CAAA;IACrD,CAAC,CAAA;IAED,IAAM,cAAc,GAAG,UAAC,CAAmC;QACzD,IAAM,IAAI,GAAG,QAAQ,CAAC,aAAa,CAAC,MAAM,CAAC,CAAA;QAC3C,QAAQ,EAAE,CAAA;QACV,SAAS,QAAQ;YACf,IAAI,CAAC,EAAE,CAAC,IAAI,EAAE,EAAE,IAAI,EAAE,OAAO,CAAC,OAAO,CAAC,MAAM,CAAC,CAAC,EAAE,CAAC,CAAC,EAAE,IAAI,EAAE,EAAE,CAAC,EAAE,yBAAyB,EAAE,EAAE,CAAC,CAAA;QAC/F,CAAC;IACH,CAAC,CAAA;IAED,kDAAkD;IAClD,kBAAkB;IAClB,oEAAoE;IACpE,kDAAkD;IAClD,mCAAmC;IAEnC,0DAA0D;IAC1D,4BAA4B;IAC5B,4BAA4B;IAC5B,wCAAwC;IACxC,yBAAyB;IACzB,oCAAoC;IACpC,wBAAwB;IACxB,4DAA4D;IAC5D,QAAQ;IACR,OAAO;IAEP,uDAAuD;IACvD,iBAAiB;IACjB,4BAA4B;IAC5B,+DAA+D;IAC/D,yGAAyG;IACzG,UAAU;IACV,OAAO;IAEP,QAAQ;IAER,OAAO,CACL,uBAAK,EAAE,EAAC,mBAAmB,EAAC,WAAW,EAAE,cAAc,EAAE,UAAU,EAAE,cAAqB,EAAE,GAAG,EAAE,GAAG,gBAClG,uBAAK,OAAO,EAAC,aAAa,EAAC,mBAAmB,EAAC,eAAe,gBAC5D,eAAM,CAAC,EAAC,yBAAyB,GAAG,IAChC,IACF,CACP,CAAA;AACH,CAAC,CAAA"}