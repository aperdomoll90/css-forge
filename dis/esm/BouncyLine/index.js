export { BouncyLine } from "./BouncyLine";
//# sourceMappingURL=index.js.map