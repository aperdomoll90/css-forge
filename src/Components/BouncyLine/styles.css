#BouncyLineWrapper {
  min-width: 100%;
  min-height: 100%;
  position: relative;
  margin: 0;
  padding: 0;
  background: #0b0b0b86;
  display: grid;
  place-content: center;
  overflow: hidden;
  z-index: 0;
  cursor: grabbing;
}
#BouncyLineWrapper svg {
  height: 100%;
  width: 100%;
  position: absolute;
  top: 0;
  left: 0;
  margin: 0;
  display: block;
}

#BouncyLineWrapper svg path {
  stroke: white;
  /* stroke: rbg(238,0,0); */
  fill: none;
}
