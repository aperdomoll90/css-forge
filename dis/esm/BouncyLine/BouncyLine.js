var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import { jsx as _jsx } from "react/jsx-runtime";
import { useRef, useState } from 'react';
import { gsap, Elastic } from 'gsap';
import './styles.css';
import '../utils/GlobalStyles.css';
export var BouncyLine = function () {
    var _a = useState({ x: 0, y: 0 }), coords = _a[0], setCoords = _a[1];
    var ref = useRef(null);
    var handleMouseMov = function (e) {
        if (ref && ref.current) {
            var width = ref.current.clientWidth;
            var height = ref.current.clientHeight;
            var offX = (e.nativeEvent.offsetX / width) * 500;
            var offY = (e.nativeEvent.offsetY / height) * 500;
            setCoords({
                x: offX,
                y: offY,
            });
            updatePath(width);
        }
    };
    var updatePath = function (width) {
        var path = document.querySelector('path');
        var x = coords.x, y = coords.y;
        path.setAttribute('d', "M250,0 Q".concat(x, ",").concat(y, " 250,500"));
    };
    var handleMouseOut = function (e) {
        var path = document.querySelector('path');
        snapBack();
        function snapBack() {
            gsap.to(path, { ease: Elastic.easeOut.config(1, 0), attr: { d: 'M250,0 Q250,250 250,500' } });
        }
    };
    // same effect but using JavaScrip Event Listeners
    // useEffect(()=>{
    //     let areaEffect = document.getElementById('BouncyLineWrapper')
    //     const path = document.querySelector('path')
    //     // let areaEffect = document
    //     areaEffect.addEventListener('mousemove', event => {
    //     let x = event.clientX
    //     let y = event.clientY
    //     let width = x / window.innerWidth
    //     updateCoords(x, y)
    //     function updateCoords(x, y) {
    //       x = width * 500
    //       path.setAttribute('d', `M250,0 Q${x},${y} 250,500`)
    //     }
    //   })
    //   areaEffect.addEventListener('mouseout', event => {
    //     snapBack()
    //     function snapBack() {
    //         // path.setAttribute('d', 'M250,0 Q250,250 250,500')
    //         gsap.to(path, { ease: Elastic.easeOut.config(1,0,3), attr: { d: 'M250,0 Q250,250 250,500' } })
    //       }
    //   })
    // },[])
    return (_jsx("div", __assign({ id: 'BouncyLineWrapper', onMouseMove: handleMouseMov, onMouseOut: handleMouseOut, ref: ref }, { children: _jsx("svg", __assign({ viewBox: '0 0 500 500', preserveAspectRatio: 'xMidYMid meet' }, { children: _jsx("path", { d: 'M250,0 Q250,250 250,500' }) })) })));
};
//# sourceMappingURL=BouncyLine.js.map