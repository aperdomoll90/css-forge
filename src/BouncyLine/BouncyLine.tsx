import { useRef, useState } from 'react'
import { gsap, Elastic } from 'gsap'
import './styles.css'

export const BouncyLine: React.FC<{}> = () => {
  const [coords, setCoords] = useState({ x: 0, y: 0 })
  const ref = useRef<HTMLDivElement>(null)

  const handleMouseMov = (e: React.MouseEvent<HTMLDivElement>) => {
    if (ref && ref.current) {
      const width = ref.current.clientWidth
      const height = ref.current.clientHeight
      const offX = (e.nativeEvent.offsetX / width) * 500
      const offY = (e.nativeEvent.offsetY / height) * 500
      setCoords({
        x: offX,
        y: offY,
      })
      updatePath(width)
    }
  }
  const updatePath = (width: number) => {
    const path = document.querySelector('path') as SVGPathElement
    let { x, y } = coords
    path.setAttribute('d', `M250,0 Q${x},${y} 250,500`)
  }

  const handleMouseOut = (e: React.MouseEvent<HTMLDivElement>) => {
    const path = document.querySelector('path')
    snapBack()
    function snapBack() {
      gsap.to(path, { ease: Elastic.easeOut.config(1, 0), attr: { d: 'M250,0 Q250,250 250,500' } })
    }
  }

  // same effect but using JavaScrip Event Listeners
  // useEffect(()=>{
  //     let areaEffect = document.getElementById('BouncyLineWrapper')
  //     const path = document.querySelector('path')
  //     // let areaEffect = document

  //     areaEffect.addEventListener('mousemove', event => {
  //     let x = event.clientX
  //     let y = event.clientY
  //     let width = x / window.innerWidth
  //     updateCoords(x, y)
  //     function updateCoords(x, y) {
  //       x = width * 500
  //       path.setAttribute('d', `M250,0 Q${x},${y} 250,500`)
  //     }
  //   })

  //   areaEffect.addEventListener('mouseout', event => {
  //     snapBack()
  //     function snapBack() {
  //         // path.setAttribute('d', 'M250,0 Q250,250 250,500')
  //         gsap.to(path, { ease: Elastic.easeOut.config(1,0,3), attr: { d: 'M250,0 Q250,250 250,500' } })
  //       }
  //   })

  // },[])

  return (
    <div id='BouncyLineWrapper' onMouseMove={handleMouseMov} onMouseOut={handleMouseOut as any} ref={ref}>
      <svg viewBox='0 0 500 500' preserveAspectRatio='xMidYMid meet'>
        <path d='M250,0 Q250,250 250,500' />
      </svg>
    </div>
  )
}
