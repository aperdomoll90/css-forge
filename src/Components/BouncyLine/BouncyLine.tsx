import { useRef, useState, useEffect } from 'react'
import { gsap, Elastic } from 'gsap'
import './styles.css'

function BouncyLine() {
  const [coords, setCoords] = useState({ x: 0, y: 0 })
  const ref = useRef<HTMLInputElement>()

  const handleMouseMov = (e: React.ChangeEvent<HTMLInputElement>) => {
    const width = ref.current.clientWidth
    const height = ref.current.clientHeight
    const offX = (e.nativeEvent.offsetX / width) * 500
    const offY = (e.nativeEvent.offsetY / height) * 500
    setCoords({
      x: offX,
      y: offY,
    })
    updatePath(width)
  }
  const updatePath = (width: number) => {
    const path = document.querySelector('path') as SVGPathElement
    let { x, y } = coords
    path.setAttribute('d', `M250,0 Q${x},${y} 250,500`)
  }

  const handleMouseOut = (e: React.ChangeEvent<HTMLInputElement>) => {
    const path = document.querySelector('path')
    snapBack()
    function snapBack() {
      gsap.to(path, { ease: Elastic.easeOut.config(1, 0, 3), attr: { d: 'M250,0 Q250,250 250,500' } })
    }
  }

  // same effect but using JavaScrip Event Listeners
  // useEffect(()=>{
  //     let areaEffect = document.getElementById('BouncyLineWrapper')
  //     const path = document.querySelector('path')
  //     // let areaEffect = document

  //     areaEffect.addEventListener('mousemove', event => {
  //     let x = event.clientX
  //     let y = event.clientY
  //     let width = x / window.innerWidth
  //     updateCoords(x, y)
  //     function updateCoords(x, y) {
  //       x = width * 500
  //       path.setAttribute('d', `M250,0 Q${x},${y} 250,500`)
  //     }
  //   })

  //   areaEffect.addEventListener('mouseout', event => {
  //     snapBack()
  //     function snapBack() {
  //         // path.setAttribute('d', 'M250,0 Q250,250 250,500')
  //         gsap.to(path, { ease: Elastic.easeOut.config(1,0,3), attr: { d: 'M250,0 Q250,250 250,500' } })
  //       }
  //   })

  // },[])

  return (
    <div id='BouncyLineWrapper' preserveAspectRatio='xMidYMid meet' onMouseMove={handleMouseMov} onMouseOut={handleMouseOut} ref={ref}>
      <svg viewBox='0 0 500 500'>
        <path d='M250,0 Q250,250 250,500' />
      </svg>
    </div>
  )
}

export default BouncyLine
