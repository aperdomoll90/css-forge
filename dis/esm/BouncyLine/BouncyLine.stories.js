import { jsx as _jsx } from "react/jsx-runtime";
import { BouncyLine } from './BouncyLine';
import StorybookContainer from '../StorybookContainer';
export default {
    title: 'User Behavior Animation',
    component: BouncyLine,
    parameters: {
        jest: ['BouncyLine.test.tsx'],
    },
};
var BouncyLineTemplate = function () { return (_jsx(StorybookContainer, { children: _jsx(BouncyLine, {}) })); };
export var BouncyLineComponent = BouncyLineTemplate.bind({});
//# sourceMappingURL=BouncyLine.stories.js.map