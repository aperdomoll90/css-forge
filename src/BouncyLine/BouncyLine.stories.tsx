import { Meta, StoryFn } from '@storybook/react'
import {BouncyLine} from './BouncyLine'
import StorybookContainer from '../StorybookContainer'


export default {
  title: 'User Behavior Animation',
  component: BouncyLine,
  parameters: {
    jest: ['BouncyLine.test.tsx'],
  },
} as Meta

const BouncyLineTemplate: StoryFn = () => (
  <StorybookContainer>
    <BouncyLine />
  </StorybookContainer>
)

export const BouncyLineComponent = BouncyLineTemplate.bind({})
