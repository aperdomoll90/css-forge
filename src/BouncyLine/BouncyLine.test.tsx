import { render } from '@testing-library/react'
import { BouncyLine } from './BouncyLine'

describe('BouncyLine component', () => {
  const renderComponent = () => render(<BouncyLine />)
  it('should match snapshots', () => {
    const { container } = renderComponent()
    expect(container).toMatchSnapshot()
  })
})
