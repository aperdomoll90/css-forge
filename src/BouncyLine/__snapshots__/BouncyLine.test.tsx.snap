// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BouncyLine component should match snapshots 1`] = `
<div>
  <div
    id="BouncyLineWrapper"
  >
    <svg
      preserveAspectRatio="xMidYMid meet"
      viewBox="0 0 500 500"
    >
      <path
        d="M250,0 Q250,250 250,500"
      />
    </svg>
  </div>
</div>
`;
